### PR TITLE
feat: added chat api tests

### PR DESCRIPTION
**Pull request by irinakartun**

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Deployment status: success

File: e2e/src/testData/api/chatApiHelper.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export class ChatApiHelper extends BaseApiHelper {
+  public async postRequest(conversation: Conversation) {
+    const requestData = {
+      modelId: conversation.model.id,
+      messages: [conversation.messages[0]],
+      id: conversation.id,
+      temperature: conversation.temperature,
+      prompt: conversation.prompt,
+      assistantModelId: conversation.assistantModelId,
+      selectedAddons: conversation.selectedAddons,
```
Comment (IlyaBondar) [inline e2e/src/testData/api/chatApiHelper.ts]:
```suggestion
      ...conversation,
      messages: [conversation.messages[0]]
```

File: e2e/src/testData/api/fileApiHelper.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  public static getContentTypeForFile(filename: string) {
+    const extension = filename.match(/(?<=\.).+$/g);
+    switch (extension![0]) {
+      case 'png':
+        return 'image/png';
+      case 'jpg':
+        return 'image/jpeg';
+    }
```
Comment (IlyaBondar) [inline e2e/src/testData/api/fileApiHelper.ts]: what's about default case i.e. for txt-file?

**Comment by mikitabut** [comment]
/deploy-review
>Deployment status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443247)

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443248)

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [running](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443251)

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443256)

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443288)

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/443341)

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: success

**Review by mikitabut — DISMISSED** [review]

File: apps/chat-e2e/src/testData/api/chatApiHelper.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export class ChatApiHelper extends BaseApiHelper {
+  public async postRequest(conversation: Conversation) {
+    const requestData = {
+      modelId: conversation.model.id,
+      messages: [conversation.messages[0]],
+      id: conversation.id,
+      temperature: conversation.temperature,
+      prompt: conversation.prompt,
+      assistantModelId: conversation.assistantModelId,
+      selectedAddons: conversation.selectedAddons,
```
Comment (IlyaBondar) [inline apps/chat-e2e/src/testData/api/chatApiHelper.ts]: You can get `conversation` as it is and change only difference (`messages` in your case). Or is there other logic?
```suggestion
      ...conversation,
	      messages: [conversation.messages[0]],
```

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-552.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**